Lokad.AzureEventStore.Cache/AzureCacheProvider.cs

```diff
@@ -51,36 +54,38 @@ public AzureCacheProvider(CloudBlobContainer container)
         ///     Returned blobs are listed by descending name (i.e. most recent
         ///     first).
         /// </remarks>
-        private async Task<CloudBlockBlob[]> Blobs(string stateName)
+        private async Task<BlobItem[]> Blobs(string stateName)
         {
             // Calling CreateIfNotExists with insufficient privilege will
             // throw an exception _even if_ the container _exists_.
             // Attempting to create the container only if it doesn't exist
             // allows to read existing state caches even if the caller does not
             // have sufficient permission to create the container.
-            if (AllowWrite && ! await _container.ExistsAsync().ConfigureAwait(false))
+            if (AllowWrite && !await _container.ExistsAsync().ConfigureAwait(false))
                 await _container.CreateIfNotExistsAsync().ConfigureAwait(false);
 
-            var result = new List<CloudBlockBlob>();
+            var result = new List<BlobItem>();
 
-            var ct = default(BlobContinuationToken);
+            var token = default(string);
             while (true)
             {
-                var list = await _container.ListBlobsSegmentedAsync(
-                    stateName,
-                    /* useFlatBlobListing: */true,
-                    BlobListingDetails.None,
-                    /* max results */ null, ct,
-                    new BlobRequestOptions(),
-                    new OperationContext());
-
-                ct = list.ContinuationToken;
-
-                foreach (var item in list.Results)
-                    if (item is CloudBlockBlob blob)
+                var list = _container.GetBlobsAsync(traits: BlobTraits.Metadata,
+                   states: BlobStates.None,
+                   prefix: stateName,
+                   cancellationToken: default(CancellationToken))
+                   .AsPages(token);
```

Comment:
My understanding is that iterating over `.AsPages(null)` will result in all blobs being enumerated, so there is no need to keep track of the continuation token, or have an outer `while` loop.

Reply:
Correct!, Done.